core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1093,18 +1093,23 @@ protected Scope visitTableExecute(TableExecute node, Optional<Scope> scope)
 
             TableExecuteHandle executeHandle =
                     metadata.getTableHandleForExecute(
-                                    session,
-                                    tableHandle,
-                                    procedureName,
-                                    tableProperties)
+                            session,
+                            tableHandle,
+                            procedureName,
+                            tableProperties,
+                            procedureMetadata.getExecutionMode().isReadsData())
                             .orElseThrow(() -> semanticException(NOT_SUPPORTED, node, "Procedure '%s' cannot be executed on table '%s'", procedureName, tableName));
 
             analysis.setTableExecuteHandle(executeHandle);
 
             analysis.setUpdateType("ALTER TABLE EXECUTE");
             analysis.setUpdateTarget(tableName, Optional.of(table), Optional.empty());
 
-            return createAndAssignScope(node, scope, Field.newUnqualified("rows", BIGINT));
+            if (procedureMetadata.getExecutionMode().isReadsData()) {
+                return createAndAssignScope(node, scope, Field.newUnqualified("rows", BIGINT));
+            }
+
+            return createAndAssignScope(node, scope, Field.newUnqualified("status", VARCHAR));
```

Comment:
I do not think we need status. Either operation completes correctly. Or we get an error, which is passed via exception.